### PR TITLE
[HttpClient] Add unit tests for `RecordException` case

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #.

## Changes

Adding few tests to validate `RecordException = true` behavior in different HttpClient and HttpWebRequest methods for both .NET and .NET Framework.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs [hunk 1]
```diff
@@ -462,6 +466,89 @@ public async Task HttpClientInstrumentationContextPropagation()
...
+
+            using var traceprovider = Sdk.CreateTracerProviderBuilder()
+                   .AddHttpClientInstrumentation(o => o.RecordException = true)
+                   .AddInMemoryExporter(exportedItems)
+                   .Build();
+
+            using var c = new HttpClient();
+            try
+            {
+                await c.GetAsync("https://www.invalidurl.com");
```
Comment (TimothyMothra) [inline test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs:483]: This is risky. 
This test may fail if someone registers this domain.

If you're just trying to have the httpclient throw an exception, I think you can use an invalid url. (eg `GetAsync("https://")` ).

If you're trying to force a network error specifically, I think you could use a non-routable ip address (eg `GetAsync("https://10.0.0.0/invalidurl")`).

Comment (vishweshbankwar) [reply test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs]: This one times out https://10.0.0.0/invalidurl. I have updated it to use something random which we use in other tests.

**Review by TimothyMothra — APPROVED** [review]
LGTM

**Review by cijothomas — APPROVED** [review]
tests LGTM!
We need to update doc to inform users about when can they except exception be recorded by the library.

**Comment by cijothomas** [comment]
Ping from https://github.com/open-telemetry/opentelemetry-dotnet/issues/2215

**Pull request merged**